docs/proposals/api-types-package-structure.md

```diff
@@ -0,0 +1,381 @@
+this api doesn't have a group.
+
+```
+apis
+└── group
+    ├── install
+    ├── types.go
+    ├── v1
+    │   ├── install
+    │   │   ├── conversion
+    │   │   │   ├── generated_conversion.go
+    │   │   │   └── manual_conversion_overrides.go
+    │   │   ├── defaulting
+    │   │   │   ├── generated_defaulters.go
+    │   │   │   └── generated_default_stubs.go
+    │   │   └── validation
+    │   │       ├── generated_validation.go
+    │   │       └── generated_validation_stubs.go
+    │   └── types.go
+    ├── v1beta1
+    │   ├── install
+    │   │   ├── conversion
+    │   │   │   ├── generated_conversion.go
+    │   │   │   └── manual_conversion_overrides.go
+    │   │   ├── defaulting
+    │   │   │   ├── generated_defaulters.go
+    │   │   │   └── generated_default_stubs.go
+    │   │   └── validation
+    │   │       ├── generated_validation.go
+    │   │       └── generated_validation_stubs.go
+    │   └── types.go
+    └── v2alpha1
+        ├── install
+        │   ├── conversion
+        │   │   ├── generated_conversion.go
+        │   │   └── manual_conversion_overrides.go
+        │   ├── defaulting
+        │   │   ├── generated_defaulters.go
+        │   │   └── generated_default_stubs.go
+        │   └── validation
+        │       ├── generated_validation.go
+        │       └── generated_validation_stubs.go
+        └── types.go
+```
+
+### Package: `apis/`
+
+Multiple groups may live in the same tree. Each group gets its own directory
+under `apis/`. We'll assume an `apis/` directory of this structure in all tools.
+
+TBD: Either test code will live here, or no code.
+
+### Package: `apis/group/`
+
+Multiple versions live in a group. Each version gets its own subdirectory.
+
+Additionally, a types.go file contains an "internal" version of the group's
+objects. This file is used as a destination/source for conversion functions.
+
+### Package: `apis/group/install`
+
+This package installs *all* of the versions of the group.
+
+### Package: `apis/group/v1`
+
+'v1' is taken to be the current stable version of this api.
+
+`types.go` contains the types for this version of the api group.
+
+This package has tightly controlled imports. It's allowed to include the
+Kubernetes unversioned API types (to be moved to `pkg/apis/unversioned`) and
+little else. In particular, it *must not* import its parent `apis/group`.
+
+### Package: `apis/group/v1/install`
+
+This package, if imported, registers the v1 version of this group, its
+conversion, validation, and defaulting functions.
+
+We have separate install packages for every version to allow deliberate import
+choices to be made.
+
+### Package: `apis/group/v1/install/conversion`
+
+This package contains conversion functions, and if imported, will register them.
+
+There are two sorts of conversion functions: those which could be automatically
+generated, and those which have manual additions. All the conversion functions
```

Comment:
Do we have any conversions for which we can't auto-generate SOMETHING?  "could be automatically generated" makes it sound as if we do.

I really like calling this out specifically as "manual overrides".  That is the pattern we should follow - I am certain there are conversions that are getting done wrongly because a new field was added and the conversion was not updated.  every conversion should start with the auto-generated and then make tweaks on that.